src/chunk.c

```diff
@@ -657,6 +657,54 @@ get_am_name_for_rel(Oid relid)
 }
 #endif
 
+static void
+copy_hypertable_acl_to_relid(Hypertable *ht, Oid relid)
+{
+	HeapTuple ht_tuple;
+	bool is_null;
+	Datum acl_datum;
+	Relation class_rel;
+
+	/* We open it here since there is no point in trying to update the tuples
+	 * if we cannot open the Relation catalog table */
+	class_rel = table_open(RelationRelationId, RowExclusiveLock);
+
+	ht_tuple = SearchSysCache1(RELOID, ObjectIdGetDatum(ht->main_table_relid));
+	Assert(HeapTupleIsValid(ht_tuple));
+
+	/* We only bother about setting the chunk ACL if the hypertable ACL is
+	 * non-null */
+	acl_datum = SysCacheGetAttr(RELOID, ht_tuple, Anum_pg_class_relacl, &is_null);
+	if (!is_null)
+	{
+		HeapTuple chunk_tuple, newtuple;
+		Datum new_val[Natts_pg_class] = { 0 };
+		bool new_null[Natts_pg_class] = { false };
+		bool new_repl[Natts_pg_class] = { false };
+		Acl *acl = DatumGetAclPCopy(acl_datum);
```

Comment:
Why do we need a copy here? Should it be freed given that this could accumulate memory on the insert path?

Reply:
Right, we don't need a copy. Remains from the first code draft. Removed.